scripts/cache.py

```diff
@@ -76,30 +76,34 @@ def read_args(update, cut_opi, export_tile):
         print("\nArguments: ", args)
 
     if not export_tile and (args.rgb is not None) and os.path.isdir(args.rgb):
-        raise SystemExit("create_cache.py: error: invalid pattern: " + args.rgb)
+        raise SystemExit("ERROR: invalid pattern: " + args.rgb)
     if not export_tile and (args.ir is not None) and os.path.isdir(args.ir):
-        raise SystemExit("create_cache.py: error: invalid pattern: " + args.ir)
+        raise SystemExit("ERROR: invalid pattern: " + args.ir)
     if not export_tile and (args.rgb is None) and (args.ir is None):
-        raise SystemExit("create_cache.py: error: no input data")
+        raise SystemExit("ERROR: no input data")
 
     if update is False:
         if os.path.isdir(args.cache) and (cut_opi or export_tile) is False:
-            raise SystemExit("Cache (" + args.cache + ") already in use")
+            raise SystemExit("ERROR: Cache (" + args.cache + ") already in use")
     else:
         if not os.path.isdir(args.cache):
-            raise SystemExit("Cache '" + args.cache + "' doesn't exist.")
+            raise SystemExit("ERROR: Cache '" + args.cache + "' doesn't exist.")
+
+    if args.table[0].isdigit():
+        raise SystemExit("ERROR: First char of table must not be a digit")
 
     if not cut_opi:
         db_graph = gdal.OpenEx(args.graph, gdal.OF_VECTOR)
         if db_graph is None:
-            raise SystemExit("Connection to database failed")
+            raise SystemExit("ERROR: Connection to database failed")
 
         # Test pour savoir si le nom de la table est correct
         if db_graph.ExecuteSQL("select * from " + args.table) is None:
-            raise SystemExit("table " + args.table + " doesn't exist")
+            raise SystemExit("ERROR: table " + args.table + " doesn't exist")
 
         if not export_tile:
-            args.x_min, args.x_max, args.y_min, args.y_max = db_graph.GetLayer().GetExtent()
+            args.x_min, args.x_max, args.y_min, args.y_max = \
+                db_graph.GetLayer(args.table).GetExtent()
```

Comment:
Régression quand on utilise un GPKG, fonctionne correctement avec une base de données.

Reply:
Fonctionnel quand la table du geopackage ne comptorte pas de chiffre comme premier caractère.

Reply:
Traceback (most recent call last):
File "D:\PackO\scripts\create_cache.py", line 6, in
cache.generate(update=False)
File "D:\PackO\scripts\cache.py", line 249, in generate
args = read_args(update, False, False)
File "D:\PackO\scripts\cache.py", line 106, in read_args
db_graph.GetLayer(args.table).GetExtent()
AttributeError: 'NoneType' object has no attribute 'GetExtent'

Il n'est dit nul part ici que j'e n'ai pas le droit de mettre un chiffre => erreur non compréhensible d'un point de vue utilisateur